src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -402,17 +468,17 @@ private unsafe bool WasEventDetected(int pollFileDescriptor, int valueFileDescri
 
             while (!cancellationToken.IsCancellationRequested)
             {
-                // poll every 1 millisecond
+                // Wait until something happens
```

Comment:
I believe we actually can't change it like this, because if we do then we won't be able to detect if a cancellation was requested, that is why we were polling every 1 ms. The idea is that if you are running a multithreaded app, you may have a different thread that will eventually say that it doesn't want to keep blocking this, so it might send a cancellation request, and if you have pollingtimeout set to 0, then this will never be detected by this thread and the execution will continue

Reply:
I don't understand. The cancellationToken is checked in the while loop above, so this will work even if the pollingtimeout is 0. This will only block forever if pollingtimeout is -1, which probably shouldn't be used. Also, the idea is to _increase_ the pollingtimeout to 100ms or so, because this reduces the time during which you are not waiting. And if an event happens while you are not inside the epoll_wait function, it won't be detected. I'll test this again, now that the code is otherwise stable.

Reply:
oh my bad I thought 0 on epoll had the same effect as -1, so that is why I said that if this blocks indefinitely then a cancellation request wouldn't get detected. I actually have never tried setting it to 0, I suppose it would never block so it just checks and then goes on.

>And if an event happens while you are not inside the epoll_wait function, it won't be detected

This I believe is actually not true. That is why we used to do the LSeek before, since even if you get an event in between running the loop but not actually on the epoll_wait function, the epoll_wait will still return true next time you call it. This we have test before to make sure we wouldn't miss events like this, and that is actually why after getting an event we were using the LSeek to make sure that we reset the event so epoll wouldn't treat it as unprocessed again.

Reply:
@joperezr It seems you're right. It doesn't actually matter what time I use, all the tests pass anyway (even if I add an extra Sleep(5)). So the timeout should probably just be selected such that it is a compromise between a fast reaction to a Cancel and extra CPU load (due to awaking the CPU and performing useless loops). I'd go for something between 50ms and 100ms. 1ms (the previous value) is to low, IMHO.

Reply:
Set to a constant of 50ms.